README.md

```diff
@@ -112,6 +112,19 @@ setTimeNextBlock: set the timestamp of the next block(does not actually mine)
 For global options help run: hardhat help
 ```
 
+This task can receive the time to be set as a number(will be interpreted as a unix timestamp, in seconds), and a string in various formats. Its hreMethod counterpart can also receive a Date object.
+
+In any case, it will fail if the date is not a second-exact date (i.e. has not zero milliseconds).
+
+String formats are thos supported by [Date.parse()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) but we recommend sticking to ISO string or YYYY-MM-DD(as the other depend on the timezone you are in and/or to implementation-specific date formats).
+
+
+Examples: 
+
+- `hardhat setTimeNextBlock 2549333405`: Will set the time of the next block to Fri Oct 14 2050 04:10:05 GMT+0000
+- `hardhat setTimeNextBlock 2050-10-14`: Will set the time of the next block to Fri Oct 14 2050 00:00:00 GMT+0000
+- `hardhat setTimeNextBlock 2050-10-14T01:41:43+00:00`: Will set the time of the next block to Fri Oct 14 2050 01:41:43 GMT+0000
```

Comment:
Is this right?

> For example, "2011-10-10" (date-only form), "2011-10-10T14:48:00" (date-time form), or "2011-10-10T14:48:00.000+09:00" (date-time form with milliseconds and time zone) can be passed and will be parsed. When the time zone offset is absent, date-only forms are interpreted as a UTC time and date-time forms are interpreted as local time.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#date_time_string_format

Reply:
Can you be more explicit on why the doc is wrong? I read that doc and couldn't find anything that contradicts with what I have written(note that the 126th line has the timezone at the end as +00:00)

Reply:
I would add the `ms` and then the timezone just to make it clear the expected format

Reply:
The timezone is there, and the ms is not in the ISO 8601 specs according to wikipedia(on top of that, the blockchain cannot handle milliseconds)